Add skybrowser settings

ylvaselling opened the pull request:
* Adds skybrowser settings
    * Note that it doesn't look great because of that the properties are not in master 
* Removes "AAS" from worldwidetelescope and removed their logo - replaced it with an icon like in missions panel

Needs testing 🙏

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/SettingsDisplayCopies.tsx:
```diff
@@ -0,0 +1,75 @@
+          viewOptions={{}}
+        />
+      </Collapsable>
+      <Group w={'100%'}>
```
```suggestion
      <Group grow>
```

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/SettingsDisplayCopies.tsx:
```diff
@@ -0,0 +1,75 @@
+          onClick={() => {
+            luaApi?.skybrowser.addDisplayCopy(id, noOfCopies, newPosition);
+          }}
+          flex={1}
```
```suggestion
```
solved by replacing 100% width with grow

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/SettingsDisplayCopies.tsx:
```diff
@@ -0,0 +1,75 @@
+          onClick={() => {
+            luaApi?.skybrowser.removeDisplayCopy(id);
+          }}
+          flex={1}
```
```suggestion
```
solved by replacing 100% width with grow

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/SettingsDisplayCopies.tsx:
```diff
@@ -0,0 +1,75 @@
+      <Property uri={`ScreenSpace.${id}.Scale`} />
+      <Property uri={`ScreenSpace.${id}.FaceCamera`} />
+      <Property uri={`ScreenSpace.${id}.UseRadiusAzimuthElevation`} />
+      <Collapsable title={'Add New Display Copy Settings'}>
```
Add some padding/margin at the bottom

WeirdRubberDuck replied:
```suggestion
      <Collapsable title={<Title order={4}>Add New Display Copy Settings</Title>}>
```
following the other suggested hierarchy

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/Settings.tsx:
```diff
@@ -1,54 +1,114 @@
+      />
+      <Property uri={`Scene.${targetId}.Renderable.ApplyRoll`} />
+      <Property uri={`ScreenSpace.${id}.PointSpacecraft`} />
+      <Collapsable title={'Display Copies'}>
+        <SettingsDisplayCopies id={id} />
+      </Collapsable>
+      <Collapsable title={'General Settings'}>
```
We need to add some hierarchy for the different sections here. First, use titles for the collapsables. 

See other code suggestions for ordering - these will be based on that this change is done
```suggestion
      <Collapsable title={<Title order={3}>Display Copies</Title>}>
        <SettingsDisplayCopies id={id} />
      </Collapsable>
      <Collapsable title={<Title order={3}>General Settings</Title>}>
```

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/SettingsDisplayCopies.tsx:
```diff
@@ -0,0 +1,75 @@
+          Remove
+        </Button>
+      </Group>
+      <Title order={2} mt={'md'} mb={'sm'}>
```
```suggestion
      <Title order={4} mt={'md'} mb={'sm'}>
```
Note that the "Settings" label is order 2, and we want this to be below that. We also want it to be below the suggested title I added for the Collapsable

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/SettingsDisplayCopies.tsx:
```diff
@@ -0,0 +1,75 @@
+        Added Display Copies
+      </Title>
+      {Object.entries(displayCopies).map(([key, entry]) => (
+        <Fragment key={key}>
+          <Property uri={`ScreenSpace.${id}.${entry.idShowProperty}`} />
+          <Property uri={`ScreenSpace.${id}.${key}`} />
+        </Fragment>
```
How about grouping these in some component that makes it clear that they belong together, visually?  The current flat list is quite hard to parse, even when grouped in a common parent element. 

![image](https://github.com/user-attachments/assets/7f195503-9edf-48b7-a0a5-e170d1430955)

It was much clearer when each were on one single line, like in the old UI. Maybe we want to make something custom for these instead of using the `Property` components?
![image](https://github.com/user-attachments/assets/7077bdbd-6beb-439a-badd-aaa3a060a5ac)

In general, I feel like the interface for the display copies may need some discussion for it to be useful

Also, I never thought I'd miss the old sliders, but for the display copies there were so much more interesting to use 🥲 Need to come up with a good way of editing vector properties..

ylvaselling replied:
I agree with you, however I think these properties have the same problem that many of our other properties have, which is the "enabled property" thing, doesn't really make sense to me to only implement a custom thing in only one place. We should implement the "enabled property" and then switch to that.

These properties need to be fixed anyway, currently they switch between cartesian and spherical coordinate systems, and there has been some discussion that people want both, so that needs another interface anyway

ylvaselling replied:
Potentially we could place the enabled property as the header in a collapse? That could be a quick fix. Still it would differ from how the UI looks in other places

WeirdRubberDuck replied:
As long as we "visually group" thw two properties for each display copies somehow, I think the current solution is good enough

WeirdRubberDuck replied:
Love the new styling!  Resolving this comment

WeirdRubberDuck reviewed (changes requested):
Seems to work just fine, but I had some comments

1. Do we maybe want to use the logo they have on their website? 

![image](https://github.com/user-attachments/assets/2d94464c-dcc1-404f-9d92-5a0211fc8b77)

2. When setting the border radius, the are outside the radius turns white. I don't think we want this behavior. It will for example be problematic when used in an otherwise dark dome setup

![image](https://github.com/user-attachments/assets/241f1e09-6d54-4888-914d-bd0aaa2c866b)

3. These inputs also have the issue of getting long "ugly" values from OpenSpace. You can however ignore this for now - I've added a feature to fix it in the scene tree branch I hop to get to merging that branch tomorrow or so
![image](https://github.com/user-attachments/assets/6f278a0c-350f-477a-8ca1-ea23ae6e60f6)

4. This settings does nothing anymore? 
![image](https://github.com/user-attachments/assets/522edc18-9621-4db2-bf8a-b5983ad8d78d)

5. This menu is in great need of some titles and text hierarchy, both to be understandable by screen readers and by the visible eye. I added some code suggestions

6. The end of the settings menu gets quite busy and hard to read. Part of it is due to the doubly displayed properties, but we should either way maybe create some type of visual grouping for the different menus. Let's discuss?

ylvaselling commented on src/panels/SkyBrowserPanel/BrowserTabs/Settings.tsx:
```diff
@@ -1,54 +1,114 @@
-      <Button variant={'outline'} color={'red'} onClick={deleteBrowser} mt={'lg'}>
-        Delete browser
-      </Button>
+      <NumericInput
```
👍 I think it would be nice to have the same look as a numeric property, is that possible?

ylvaselling commented:
> Seems to work just fine, but I had some comments
> 
>     1. Do we maybe want to use the logo they have on their website?
> 
> 
> ![image](https://private-user-images.githubusercontent.com/8808894/422265198-2d94464c-dcc1-404f-9d92-5a0211fc8b77.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5MzUxMjIsIm5iZiI6MTc0MTkzNDgyMiwicGF0aCI6Ii84ODA4ODk0LzQyMjI2NTE5OC0yZDk0NDY0Yy1kY2MxLTQwNGYtOWQ5Mi01YTAyMTFmYzhiNzcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTRUMDY0NzAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjkzZWJkMzA0NDk5ZDRiZDY3MWE5Y2E5ZGQ0NTUzNGRhYWI5ZGNjNDI2YmJjMzQwMjVhYjQ2YTc5Y2FhOWM0YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.6e9Zyug6nwOmS8-mjU7D-YQ4oh4-BgRiA7QYsTjfJKQ)
> 
>     2. When setting the border radius, the are outside the radius turns white. I don't think we want this behavior. It will for example be problematic when used in an otherwise dark dome setup
> 
> 
> ![image](https://private-user-images.githubusercontent.com/8808894/422270043-241f1e09-6d54-4888-914d-bd0aaa2c866b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5MzUxMjIsIm5iZiI6MTc0MTkzNDgyMiwicGF0aCI6Ii84ODA4ODk0LzQyMjI3MDA0My0yNDFmMWUwOS02ZDU0LTQ4ODgtOTE0ZC1iZDBhYWEyYzg2NmIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTRUMDY0NzAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWY4YzUzZTMxMWZhMmEyZmZkNDg1ZDYwZGVhYTZlY2UyYzc5MjlkMTA0NzkzYjQ0Y2NiNDhjNDA3NGY2NDg1OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.oNDModLrJIXJapna1_UFUM5Yym4X3qJfMVxE3SlmgWc)
> 
>     3. These inputs also have the issue of getting long "ugly" values from OpenSpace. You can however ignore this for now - I've added a feature to fix it in the scene tree branch I hop to get to merging that branch tomorrow or so
>        ![image](https://private-user-images.githubusercontent.com/8808894/422270554-6f278a0c-350f-477a-8ca1-ea23ae6e60f6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5MzUxMjIsIm5iZiI6MTc0MTkzNDgyMiwicGF0aCI6Ii84ODA4ODk0LzQyMjI3MDU1NC02ZjI3OGEwYy0zNTBmLTQ3N2EtOGNhMS1lYTIzYWU2ZTYwZjYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTRUMDY0NzAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmI2OGNlMTFmZmI0ZGViZDIzYTE2NDg4NTE1MjhmZGI2NzdjNmRhYzNlZGY5MDYxZTQ2ZDM1NDY3ZGRkZmQ4NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.WoS2QfFUTPWmj6qYiL9qgBNGxleXydkCTX32RjGWnvI)
> 
>     4. This settings does nothing anymore?
>        ![image](https://private-user-images.githubusercontent.com/8808894/422277468-522edc18-9621-4db2-bf8a-b5983ad8d78d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5MzUxMjIsIm5iZiI6MTc0MTkzNDgyMiwicGF0aCI6Ii84ODA4ODk0LzQyMjI3NzQ2OC01MjJlZGMxOC05NjIxLTRkYjItYmY4YS1iNTk4M2FkOGQ3OGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTRUMDY0NzAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzkwMGExMzUyNGE5ZmY3NDU3NDFhMjc1OTFlNTUyOGRkZGM4N2NlN2Y0MWMzZmRmYzYyNzJiYTNhZmIwZDZmMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.4t-7wtBfz8EcjBZuYIOrRhyxsvaN7QmSOG0rl_pb5pw)
> 
>     5. This menu is in great need of some titles and text hierarchy, both to be understandable by screen readers and by the visible eye. I added some code suggestions
> 
>     6. The end of the settings menu gets quite busy and hard to read. Part of it is due to the doubly displayed properties, but we should either way maybe create some type of visual grouping for the different menus. Let's discuss?
> 
>     7. It also looks like this setting still needs to be implemented:
>        ![image](https://private-user-images.githubusercontent.com/8808894/422285262-c62eca8f-75aa-4ebf-880d-a7770f3aabae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5MzUxMjIsIm5iZiI6MTc0MTkzNDgyMiwicGF0aCI6Ii84ODA4ODk0LzQyMjI4NTI2Mi1jNjJlY2E4Zi03NWFhLTRlYmYtODgwZC1hNzc3MGYzYWFiYWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTRUMDY0NzAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjMyNTExZGU1MWY1NjEwODA4ZDljN2IyMGNkOWU0NTI5NmNkY2ExOGMxNzVhYWQzNDI5NWI2M2RkNjk5NzIzYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.gj8pxKXWtYkGVvaz3RS2HsF7sHxtHi0MId18q1k-G-s)



> Seems to work just fine, but I had some comments
> 
>     1. Do we maybe want to use the logo they have on their website?
> 
> 
> ![image](https://private-user-images.githubusercontent.com/8808894/422265198-2d94464c-dcc1-404f-9d92-5a0211fc8b77.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5MzUxMjIsIm5iZiI6MTc0MTkzNDgyMiwicGF0aCI6Ii84ODA4ODk0LzQyMjI2NTE5OC0yZDk0NDY0Yy1kY2MxLTQwNGYtOWQ5Mi01YTAyMTFmYzhiNzcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTRUMDY0NzAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MjkzZWJkMzA0NDk5ZDRiZDY3MWE5Y2E5ZGQ0NTUzNGRhYWI5ZGNjNDI2YmJjMzQwMjVhYjQ2YTc5Y2FhOWM0YiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.6e9Zyug6nwOmS8-mjU7D-YQ4oh4-BgRiA7QYsTjfJKQ)
> 
>     2. When setting the border radius, the are outside the radius turns white. I don't think we want this behavior. It will for example be problematic when used in an otherwise dark dome setup
> 
> 
> ![image](https://private-user-images.githubusercontent.com/8808894/422270043-241f1e09-6d54-4888-914d-bd0aaa2c866b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5MzUxMjIsIm5iZiI6MTc0MTkzNDgyMiwicGF0aCI6Ii84ODA4ODk0LzQyMjI3MDA0My0yNDFmMWUwOS02ZDU0LTQ4ODgtOTE0ZC1iZDBhYWEyYzg2NmIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTRUMDY0NzAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWY4YzUzZTMxMWZhMmEyZmZkNDg1ZDYwZGVhYTZlY2UyYzc5MjlkMTA0NzkzYjQ0Y2NiNDhjNDA3NGY2NDg1OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.oNDModLrJIXJapna1_UFUM5Yym4X3qJfMVxE3SlmgWc)
> 
>     3. These inputs also have the issue of getting long "ugly" values from OpenSpace. You can however ignore this for now - I've added a feature to fix it in the scene tree branch I hop to get to merging that branch tomorrow or so
>        ![image](https://private-user-images.githubusercontent.com/8808894/422270554-6f278a0c-350f-477a-8ca1-ea23ae6e60f6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5MzUxMjIsIm5iZiI6MTc0MTkzNDgyMiwicGF0aCI6Ii84ODA4ODk0LzQyMjI3MDU1NC02ZjI3OGEwYy0zNTBmLTQ3N2EtOGNhMS1lYTIzYWU2ZTYwZjYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTRUMDY0NzAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZmI2OGNlMTFmZmI0ZGViZDIzYTE2NDg4NTE1MjhmZGI2NzdjNmRhYzNlZGY5MDYxZTQ2ZDM1NDY3ZGRkZmQ4NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.WoS2QfFUTPWmj6qYiL9qgBNGxleXydkCTX32RjGWnvI)
> 
>     4. This settings does nothing anymore?
>        ![image](https://private-user-images.githubusercontent.com/8808894/422277468-522edc18-9621-4db2-bf8a-b5983ad8d78d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5MzUxMjIsIm5iZiI6MTc0MTkzNDgyMiwicGF0aCI6Ii84ODA4ODk0LzQyMjI3NzQ2OC01MjJlZGMxOC05NjIxLTRkYjItYmY4YS1iNTk4M2FkOGQ3OGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTRUMDY0NzAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzkwMGExMzUyNGE5ZmY3NDU3NDFhMjc1OTFlNTUyOGRkZGM4N2NlN2Y0MWMzZmRmYzYyNzJiYTNhZmIwZDZmMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.4t-7wtBfz8EcjBZuYIOrRhyxsvaN7QmSOG0rl_pb5pw)
> 
>     5. This menu is in great need of some titles and text hierarchy, both to be understandable by screen readers and by the visible eye. I added some code suggestions
> 
>     6. The end of the settings menu gets quite busy and hard to read. Part of it is due to the doubly displayed properties, but we should either way maybe create some type of visual grouping for the different menus. Let's discuss?
> 
>     7. It also looks like this setting still needs to be implemented:
>        ![image](https://private-user-images.githubusercontent.com/8808894/422285262-c62eca8f-75aa-4ebf-880d-a7770f3aabae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5MzUxMjIsIm5iZiI6MTc0MTkzNDgyMiwicGF0aCI6Ii84ODA4ODk0LzQyMjI4NTI2Mi1jNjJlY2E4Zi03NWFhLTRlYmYtODgwZC1hNzc3MGYzYWFiYWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMTRUMDY0NzAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjMyNTExZGU1MWY1NjEwODA4ZDljN2IyMGNkOWU0NTI5NmNkY2ExOGMxNzVhYWQzNDI5NWI2M2RkNjk5NzIzYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.gj8pxKXWtYkGVvaz3RS2HsF7sHxtHi0MId18q1k-G-s)

1. Don't really know if that logo is valid - just went by what Megan said, to remove the logo. Maybe check with her if we can use the logo you're suggesting?
2. Yeah, making a ticket for it in notion. Not part of this pr - think it needs some attention on the server
3. -
4. ✅ True, I guess we can just remove that property in the engine.
5. ✅ 
6. Sure, yeah it was a bit hard to tell with the double properties so I just left it as it was in the old ui
7. True! making a ticket for it

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/SettingsDisplayCopies.tsx:
```diff
@@ -0,0 +1,119 @@
+
+      <Collapsable
+        title={
+          <>
+            Added Display Copies{' '}
+            <Badge variant={'default'}>{Object.keys(displayCopies).length}</Badge>
+          </>
```
```suggestion
          <Group>
            Added Display Copies
            <Badge variant={'default'}>{Object.keys(displayCopies).length}</Badge>
          </Group>
```

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/SettingsDisplayCopies.tsx:
```diff
@@ -0,0 +1,119 @@
+        </Button>
+        <SettingsPopout flex={0} popoverWidth={300} title={'Add Display Copy Settings'}>
+          <Box p={'sm'}>
+            <Group>
+              <InputLabel fw={'normal'}>
+                <Text span size={'sm'}>
+                  Number of copies
+                </Text>
+              </InputLabel>
+              <InfoBox>
+                {`This sets the position of the first copy. The additional copies will be evenly
+          spread out on the Azimuth, if isUsingRadiusAzimuthElevation is enabled,
+          otherwise it will spread on the Y axis.`}
+              </InfoBox>
+            </Group>
+            <NumericInput value={noOfCopies} onEnter={setNoOfCopies} />
```
1. The infobox text here is wrong
2. Is this setting even needed?? It's so easy to just click the "Add" button two times

WeirdRubberDuck replied:
But I guess it is since people want to automatically spread out multiple copies at once?  I personally find this feature quite confusing. Do we still need it? If so, the settings menu might need a bit more explaining of what the purpose of these settings are

Let's discuss, maybe?

ylvaselling replied:
What is wrong about the text?

I'm just making the ui look like before 😄 I've had similar thoughts about this setting but discussed removing it with alex and we ended up having it the way it was.

WeirdRubberDuck replied:
Ok, sure, if you've already discussed it leave it like it is. Was only a suggestion :) 

But the text for the infobox is a copy of the one that describes the "Position of the first copy". This should describe the "Number of copies" setting 😉

ylvaselling replied:
Ah, good catch. 

I'm happy make to discuss ways how to make this interface better, but I think there are plenty of things in the skybrowser that can be improved and I would like to merge this PR, so maybe we can do an overhaul another time :)

engbergandreas reviewed (commented):
1. Do we want to have min/max values shown for the Border Radius setting - including the slider? 
2. Setting the Declination to 90 or -90 will change the Right Ascension as well. To reproduce: set Right Ascension to 30, change Declination to 90 => Right ascension will be set to ~170

WeirdRubberDuck commented on src/components/SettingsPopout/SettingsPopout.tsx:
```diff
@@ -1,17 +1,30 @@
 import { SettingsIcon } from '@/icons/icons';
 
-export function SettingsPopout({ children }: PropsWithChildren) {
+interface Props {
+  popoverWidth?: number;
+  title?: string;
+}
+
+export function SettingsPopout({
+  popoverWidth,
+  title,
+  children,
+  ...styleProps
+}: Props & PropsWithChildren & MantineStyleProps) {
```
I'm not necessarily against this way of doing this, but please let's stick to the way we've done it in all other components
```suggestion
interface Props extends MantineStyleProps, PropsWithChildren {
  popoverWidth?: number;
  title?: string;
}

export function SettingsPopout({ popoverWidth, title, children, ...styleProps }: Props) {
```

WeirdRubberDuck reviewed (changes requested):
Loving the new look!  Have some thing sin the code I would like fixed, though :) 

Also, can we please fix this issue with the border radius: 
https://github.com/user-attachments/assets/dada2e0a-59fc-4c68-ab8f-c9e4c6f5d7fc



There is also an issue with the RA changing when the declination is edited, but I assume this might be harder to fix?
https://github.com/user-attachments/assets/ea478d4f-3e14-4309-a5bf-e7c093295571

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/Settings.tsx:
```diff
@@ -1,54 +1,193 @@
+              onInput={setVerticalFov}
+              flex={2}
+            />
+            <NumericInput
```
I suggest you set a ` decimalScale={x}` on these inputs.  Maybe limiting it to 3-4 decimals?

WeirdRubberDuck replied:
Goes for RA and Dec as well

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/Settings.tsx:
```diff
@@ -1,54 +1,193 @@
+          <Text size={'sm'} mb={'xs'} mt={'sm'} fw={'500'}>
+            Right Ascension
+          </Text>
+          <Group grow my={'sm'}>
```
```suggestion
          <Group grow my={'xs'}>
```

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/Settings.tsx:
```diff
@@ -1,54 +1,193 @@
+              max={90}
+              clampBehavior={'strict'}
+              onEnter={setDeclination}
+              pb={'sm'}
```
```suggestion
```

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/BrowserTabs/Settings.tsx:
```diff
@@ -1,54 +1,193 @@
+            <Tabs.Tab value={'General'}>General</Tabs.Tab>
+          </Tooltip>
+        </Tabs.List>
+        <Tabs.Panel value={'Settings'} py={'sm'}>
```
Now you add a `py={'sm'}` to all panels individually. This easily becomes out of sync

How about, instead, wrapping all the `Tabs.Panel` components in a box or something that adds that padding to all of them?

ylvaselling commented on src/panels/SkyBrowserPanel/BrowserTabs/SettingsDisplayCopies.tsx:
```diff
@@ -0,0 +1,119 @@
+              </InfoBox>
+            </Group>
+            <Box pb={'sm'}>
+              <DefaultView
```
Actually I would like all our Properties to just be wrappers of input components :) I think that would be awesome as that could unify the look of our UI

WeirdRubberDuck replied:
Yeah that would definitely be something to strive for. Maybe we can do another rewrite at some point

Have started poking a bit on the precision issue and had a similar thought

ylvaselling commented on src/panels/SkyBrowserPanel/BrowserTabs/SettingsDisplayCopies.tsx:
```diff
@@ -0,0 +1,119 @@
+          Add
+        </Button>
+        <Button
+          onClick={() => {
+            luaApi?.skybrowser.removeDisplayCopy(id);
+          }}
```
Technically these are not the same though. The lambda function will return the value, whereas the curly braces will not return anything. In this particular case I don't think it makes a difference since the functions return void but having some global setting for this would be dangerous I think

WeirdRubberDuck replied:
Ok sure, I agree with your point on the global setting. 

This is, however, how we have written essentially all of our onclick function call, and I think we should be consistent. It is also a lot cleaner looking :)

ylvaselling commented:
> Loving the new look! Have some thing sin the code I would like fixed, though :)
> 
> Also, can we please fix this issue with the border radius: https://github.com/user-attachments/assets/d3545147-2f17-4edf-9ae1-cce02fc0edd4
> 
> There is also an issue with the RA changing when the declination is edited, but I assume this might be harder to fix? https://github.com/user-attachments/assets/9c034b30-6e02-4e1b-9dc8-cefc706cae7f

The links you posted lead to 404 so not sure what you mean :)

ylvaselling commented:
The declination to -90 is a bug in the engine that was present in the old ui as well, so I don't think that is in the scope of this pr.

WeirdRubberDuck commented on src/panels/SkyBrowserPanel/WorldWideTelescope/WorldWideTelescopeView.tsx:
```diff
@@ -112,7 +115,11 @@ export function WorldWideTelescopeView() {
-        style={{ borderWidth: 0, pointerEvents: 'none' }}
+        style={{
+          border: '0px solid transparent',
+          borderRadius: radius * 1000,
```
AH, this might be what's causing the weird behavior I sent a video for

ylvaselling replied:
Yeah for some reason the background of the transparent iframe in our new ui is white. In our previous ui it was transparent. I tried to remedy it this hacky way but it doesn't work very well... the proper solution is to fix this behavior on the server which is another repo. I think for now I'll leave the background white and do a new ticket for the radius

WeirdRubberDuck commented:
> > Loving the new look! Have some thing sin the code I would like fixed, though :)
> > Also, can we please fix this issue with the border radius: https://github.com/user-attachments/assets/d3545147-2f17-4edf-9ae1-cce02fc0edd4
> > There is also an issue with the RA changing when the declination is edited, but I assume this might be harder to fix? https://github.com/user-attachments/assets/9c034b30-6e02-4e1b-9dc8-cefc706cae7f
> 
> The links you posted lead to 404 so not sure what you mean :)

Don't know what happened with the videos there... Updated so the first video should show now. Something broke when I uploaded both in one message

Here's the second one, showing how changing declination leads to some odd (? are they?) RA changes. But as I said, not sure if this is something we can fix? 
https://github.com/user-attachments/assets/1a0f593a-4f65-46fb-bc12-245692376326

ylvaselling commented:
> > > Loving the new look! Have some thing sin the code I would like fixed, though :)
> > > Also, can we please fix this issue with the border radius: https://github.com/user-attachments/assets/d3545147-2f17-4edf-9ae1-cce02fc0edd4
> > > There is also an issue with the RA changing when the declination is edited, but I assume this might be harder to fix? https://github.com/user-attachments/assets/9c034b30-6e02-4e1b-9dc8-cefc706cae7f
> > 
> > 
> > The links you posted lead to 404 so not sure what you mean :)
> 
> Don't know what happened with the videos there... Updated so the first video should show now. Something broke when I uploaded both in one message
> 
> Here's the second one, showing how changing declination leads to some odd (? are they?) RA changes. But as I said, not sure if this is something we can fix?
> 2025-03-26.16-39-52.mov

See above comment:)

WeirdRubberDuck reviewed (approved):
Looks great with the new changes!  Love the tabs in the settings. 

One thing - it would be good if the settings view is loading while the rest of the skybrowser menus are still loading. Now the settings are accessible before they are really working
